modules/grizzly/src/main/java/org/glassfish/grizzly/threadpool/VirtualThreadExecutorService.java

```diff
@@ -0,0 +1,110 @@
+package org.glassfish.grizzly.threadpool;
+
+import org.glassfish.grizzly.Grizzly;
+import org.glassfish.grizzly.localization.LogMessages;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author mazhen
+ */
+public class VirtualThreadExecutorService extends AbstractExecutorService implements Thread.UncaughtExceptionHandler {
+
+    private static final Logger logger = Grizzly.logger(VirtualThreadExecutorService.class);
+
+    private final ExecutorService internalExecutorService;
+    private Semaphore poolSemaphore;
+
+    public static VirtualThreadExecutorService createInstance() {
+        return createInstance(ThreadPoolConfig.defaultConfig().setMaxPoolSize(-1).setPoolName("Grizzly-virt-"));
+    }
+
+    public static VirtualThreadExecutorService createInstance(ThreadPoolConfig cfg) {
+        Objects.requireNonNull(cfg);
+        return new VirtualThreadExecutorService(cfg);
+    }
+
+    protected VirtualThreadExecutorService(ThreadPoolConfig cfg) {
+        internalExecutorService = Executors.newThreadPerTaskExecutor(getThreadFactory(cfg));
+        if (cfg.getMaxPoolSize() > 0) {
+            poolSemaphore = new Semaphore(cfg.getMaxPoolSize());
+        } else {
+            poolSemaphore = new Semaphore(Integer.MAX_VALUE);
```

Comment:
The Semaphore should be constructed with the `true` value for the second parameter to enable fairness - guarantee the order of tasks.